taskcluster/ci/csharp-build/kind.yml

```diff
@@ -0,0 +1,37 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+---
+loader: glean_taskgraph.loader.build_config:loader
+
+transforms:
+  - glean_taskgraph.transforms.module_build:transforms
+  - taskgraph.transforms.job:transforms
+  - taskgraph.transforms.task:transforms
+
+job-defaults:
+  attributes:
+    run-on-pr-type: full-ci
+  run-on-tasks-for: [github-pull-request, github-push]
+  description: "{module_name} - Build and test"
+  scopes:
+    - project:releng:services/tooltool/api/download/internal
+  worker-type: b-linux
+  worker:
+    chain-of-trust: true
+    docker-image: { in-tree: csharp }
+    max-run-time: 1800
+  run:
+    pre-dotnet:
+      # XXX: scripts subshell at runtime so we need to source this here
+      # to be able to access `rustup` and `rustc` from within the cc script
+      # and the gradle command. Another options could be to set those env vars
+      # here like: [export, 'PATH=$HOME/.cargo/bin:$PATH'
+      - [source, taskcluster/scripts/rustup-setup.sh]
+      - [source, taskcluster/scripts/cross-compile-setup.sh]
+      - [bash, '-c', 'echo "rust.targets=x86_64,x86,darwin,linux-x86-64,win32-x86-64-gnu\n" > local.properties']
```

Comment:
@badboy this only makes sense when running gradle, right? I guess I should add something similar to the C# project or manually run `cargo build --target ...` here, right?

Reply:
Yup, local.properties is only used by Gradle, and the rust-android-gradle plugin uses that `rust.targets` parameter to determine what it needs to pass to `cargo build` internally).

As the dotnet build system doesn't handle building the rust part yet, we need to manually invoke the individual `cargo build` commands with the right `--target` parameters.

Reply:
And I _think_ we might actually need to set up the compilers/linkers and flags in `.cargo/config` for this, e.g. the extra flags requires for a macOS build: https://github.com/mozilla/glean/blob/main/taskcluster/scripts/cross-compile-setup.sh#L8